### PR TITLE
BS-13159 illegal character in diagram name

### PR DESCRIPTION
**Pull request by rbioteau**
- Improve file name validator
- Refactor Dialog and add unit tests on validation.

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.application/src/org/bonitasoft/studio/application/actions/SaveCommandHandler.java [hunk 1]
```diff
@@ -300,17 +301,13 @@ private MainProcess getOldProcess(final MainProcess proc) {
 
     @Override
     public boolean isEnabled() {
-        final boolean res = super.isEnabled() || isDirty() ;
-        return res ;
+        final boolean res = super.isEnabled() || isDirty();
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.application/src/org/bonitasoft/studio/application/actions/SaveCommandHandler.java]: en profiter pour  enlever cette variable intermédiaire?

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.application/src/org/bonitasoft/studio/application/actions/SaveCommandHandler.java]: J'ai juste fait un ctrl+shift+o...

File: bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/DiagramUnicityValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                new WritableValue("1.0", String.class),
+                Sets.newHashSet("MyDiagram-1.0.proc"));
+
+        final IStatus status = validator.validate();
+
+        assertThat(status).isOK();
+    }
+
+    @Test
+    public void should_fail_if_diagram_name_and_versio_nexists() throws Exception {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/DiagramUnicityValidatorTest.java]: typo versio

File: bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/MustUpdateValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        final IStatus status = mustUpdateValidator.validate();
+
+        StatusAssert.assertThat(status).isOK();
+    }
+
+    @Test
+    public void should_fail_if_neither_name_nor_version_is_changed() throws Exception {
+        final MustUpdateValidator mustUpdateValidator = new MustUpdateValidator(aPool().withName("Pool").withVersion("1.0").build(), new WritableValue("Pool",
+                String.class), new WritableValue("1.0",
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/MustUpdateValidatorTest.java]: the linebreak is not the best choice

File: bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        final OpenNameAndVersionDialog dialog = new OpenNameAndVersionDialog(realmWithDisplay.getShell(), aMainProcess().withName("MyDiagram")
+                .withVersion("1.0").build(), diagramStore);
+
+        final UpdateValueStrategy nameStrategy = dialog.diagramNameUpdateStrategy();
+
+        StatusAssert.assertThat(nameStrategy.validateAfterGet("toooooooooooooooooooooooooooooooooooooooooooooooLong")).isNotOK();
+    }
+
+    @Test
+    public void should_diagram_version_strategy_validate_fileName() throws Exception {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java]: What is the difference with should_diagram_name_strategy_validate_fileName?

File: bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import org.eclipse.core.databinding.UpdateValueStrategy;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OpenNameAndVersionDialogTest {
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java]: Shouldn't we have also a test with the other case mentioned by the bug "My \\ Diagram"?

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java]: Spaces are valid characters. Backslash are not. There is no specificity about using both.

Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.diagram/src-test/java/org/bonitasoft/studio/common/diagram/dialog/OpenNameAndVersionDialogTest.java]: seems legit ^^

**Pull request merged**